src/main/java/com/book/booksstore/service/SoppingCartServiceImpl.java

```diff
@@ -0,0 +1,146 @@
+                        .equals(bookId))
+                .findFirst()
+                .orElseGet(() -> {
+                    CartItem cartItem = new CartItem();
+                    Book book = bookRepository.findById(bookId)
+                            .orElseThrow(() -> new EntityNotFoundException(
+                                    "Book not found: " + bookId));
+                    cartItem.setBook(book);
+
+                    return cartItem;
+                });
+
+        item.setQuantity(item.getQuantity() + quantity);
+        itemRepository.save(item);
+
+        return shoppingCartMapper.toDto(cart);
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartResponseDto updateItemQuantity(
+            Long cartItemId,
+            int quantity) {
+        CartItem item = itemRepository.findById(cartItemId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "CartItem not found: " + cartItemId));
+
+        ShoppingCart cart = item.getShoppingCart();
+        if (quantity <= 0) {
+            cart.getCartItems().remove(item);
+            itemRepository.delete(item);
+        } else {
+            item.setQuantity(quantity);
+            itemRepository.save(item);
+        }
+        return shoppingCartMapper.toDto(cart);
+    }
+
+    @Override
+    public void createShoppingCart(Long userId) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "User not found: " + userId));
+        ShoppingCart existing = shoppingCartRepository
+                .findFirstByUserAndIsDeletedFalse(user);
+        if (existing != null) {
+            shoppingCartMapper.toDto(existing);
+            return;
+        }
+        ShoppingCart cart = new ShoppingCart();
+        cart.setUser(user);
+        cart = shoppingCartRepository.save(cart);
+
+        shoppingCartMapper.toDto(cart);
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartResponseDto deleteItemFromShoppingCart(Long cartItemId) {
+        User user = resolveCurrentUser();
+        ShoppingCart cart = shoppingCartRepository
+                .findFirstByUserAndIsDeletedFalse(user);
+        if (cart == null) {
+            return shoppingCartMapper.toDto(new ShoppingCart());
+        }
+
+        boolean removed = cart.getCartItems().removeIf(cartItem ->
+                cartItem.getId().equals(cartItemId)
+        );
+
+        if (removed) {
+            shoppingCartRepository.save(cart);
+        }
+
+        return shoppingCartMapper.toDto(cart);
+    }
+
+    private User resolveCurrentUser() {
+        String username = SecurityContextHolder
+                .getContext()
+                .getAuthentication()
+                .getName();
+
+        UserDetails userDetails = userDetailsService
+                .loadUserByUsername(username);
+        return userRepository.findByEmail(userDetails.getUsername())
```

Comment:
Do you really need this query to the DB for each request? Is it enough to validate token?